src/main/java/de/tum/in/www1/artemis/service/connectors/jenkins/JenkinsBuildLogParseUtils.java

```diff
@@ -0,0 +1,81 @@
+package de.tum.in.www1.artemis.service.connectors.jenkins;
+
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.util.ArrayList;
+import java.util.LinkedList;
+import java.util.List;
+
+import org.apache.commons.lang.StringUtils;
+import org.jsoup.nodes.Element;
+import org.jsoup.nodes.Node;
+import org.jsoup.nodes.TextNode;
+
+import de.tum.in.www1.artemis.domain.BuildLogEntry;
+
+public class JenkinsBuildLogParseUtils {
+
+    // Pattern of the DateTime that is included in the logs received from Jenkins
+    private static final DateTimeFormatter LOG_DATA_TIME_FORMATTER = DateTimeFormatter.ofPattern("yyyy-MM-dd'T'HH:mm:ssX");
+
+    public static List<BuildLogEntry> parseBuildLogsFromJenkinsLogs(List<String> logLines) {
+        final List<BuildLogEntry> buildLogs = new ArrayList<>();
+        for (final var logLine : logLines) {
+            // The build logs that we are interested in are the ones that start with a timestamp
+            // of format [timestamp] ...
+            final String possibleTimestamp = StringUtils.substringBetween(logLine, "[", "]");
+            if (possibleTimestamp == null) {
+                continue;
+            }
+
+            try {
+                final ZonedDateTime timestamp = ZonedDateTime.parse(possibleTimestamp);
+                final String log = logLine.substring(possibleTimestamp.length() + 2);
```

Comment:
The magic **2** is very specific. Do you reckon, maybe a line comment would make it more readable ?